File: Methods/Methods-Python/find_median.py
def find_median(a):
    k = len(a) / 2 + 1
    minIndex = 0
    minValue = a[0]
    for i in range(0, k):
        minIndex = i
        minValue = a[i]

        for j in range(i + 1, len(a)):
            if a[j] < minValue:
                minIndex = j
                minValue = a[j]

        temp = a[i]
        a[i] = a[minIndex]
        a[minIndex] = temp
    if len(a) % 2 == 0:
        return (a[(len(a) / 2 - 1)] + a[len(a) / 2]) / 2
    else:
        return a[len(a)/2]

File: Methods/Methods-Java/Java source-code/find_median.java


public class find_median {
    public static double find_median_m(Integer[] a) {
        int k = a.length / 2 + 1;
        int minIndex = 0;
        int minValue = a[0];
        for (int i = 0; i < k; i++) {
            minIndex = i;
            minValue = a[i];
            for (int j = i + 1; j < a.length; j++) {
                if (a[j] < minValue) {
                    minIndex = j;
                    minValue = a[j];
                }
            }
            int temp = a[i];
            a[i] = a[minIndex];
            a[minIndex] = temp;
        }
        if (a.length % 2 == 0) {
            return (double) (a[a.length / 2 - 1] + a[a.length / 2]) / 2;
        } else {
            return a[a.length / 2];
        }
    }
}